### DISCUSSION TITLE
Assets: Image/Video/Audio Handling Methods ?

### DISCUSSION THREAD

**Discussion by jktan0504** [devlikeapro/waha#349] [Q&A]
Hi, experts and authors,

How can I download the image url from webook then save into my own storage (either local or s3)

I try use the url: http://localhost:3000/api/files/3A99332561E07E4EFD21.jpeg
either from web browser or code:
```
{"statusCode":401,"message":"Unauthorized"}
```

PS: i have whatsapp plus membership

Please help

[![patron:PLUS](https://img.shields.io/badge/patron-PLUS-a0e6ba)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by devlikepro** [comment]
Hi! Did you setup the API Key authorization? https://waha.devlike.pro/docs/how-to/security/#set-api-key
If yes - remember to add api key to your request (or use some browser extension that adds custom headers if you want to see it on the browser)

We could save it directly to S3, if you wish that feature - feel free to create an issue, so we can see how many people are interested in that and prioritize it properly! 
https://waha.devlike.pro/docs/how-to/storages/#media---s3

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by devlikepro** [comment]
Remember that there's a lifetime for files (by default 3 minutes). You can adjust that by setting `WHATSAPP_FILES_LIFETIME`
- https://waha.devlike.pro/docs/how-to/storages/#save-media-files-between-the-container-restarts
- https://waha.devlike.pro/docs/how-to/config/#files
- 
[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by ridan-genius** [comment]
it support download media using noweb or just using webjs can store media using db or local folders

[![patron:PLUS](https://img.shields.io/badge/patron-PLUS-a0e6ba)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)
